### PR TITLE
Struqture-qutip interface for SpinOperator

### PR DESCRIPTION
**Pull request by dvy-gupta**
Added function for converting SpinOperator (struqture) to qutip object along with unittests.

### CODE REVIEW

File: struqture_qutip_interface/qutip_interface.py [hunk 1]
```diff
@@ -156,6 +156,28 @@ def qobj(system: spins.SpinHamiltonianSystem, endianess: str = "little") -> qt.Q
...
+        Args:
+            spin_op: The spin system
+            endianess: first qubit to the right (little) or left (big)
+
+        Returns:
+            qt.Qobj: The QuTiP representation
+
+        """
+        number_spins: int
+        number_spins = spin_op.number_spins()
```
Comment (kbarkhqs) [inline struqture_qutip_interface/qutip_interface.py]: You should be able to consolidate these two lines into one

File: struqture_qutip_interface/qutip_interface.py [hunk 2]
```diff
@@ -156,6 +156,28 @@ def qobj(system: spins.SpinHamiltonianSystem, endianess: str = "little") -> qt.Q
...
+        spin_operator = 0
```
Comment (kbarkhqs) [inline struqture_qutip_interface/qutip_interface.py]: This needs a type annotation

File: struqture_qutip_interface/qutip_interface.py [hunk 3]
```diff
@@ -156,6 +156,28 @@ def qobj(system: spins.SpinHamiltonianSystem, endianess: str = "little") -> qt.Q
...
+    @staticmethod
+    def qobj(spin_op: spins.SpinSystem, endianess: str = "little") -> qt.Qobj:
+        r"""Returns a QuTiP representation.
+
+        Args:
+            spin_op: The spin system
+            endianess: first qubit to the right (little) or left (big)
+
+        Returns:
+            qt.Qobj: The QuTiP representation
```
Comment (kbarkhqs) [inline struqture_qutip_interface/qutip_interface.py]: This docstring seems to be identical to the one above, one of them should be changed to reflect the difference between the two. Additionally, should they have different names?

File: struqture_qutip_interface/qutip_interface.py [hunk 4]
```diff
@@ -156,6 +156,28 @@ def qobj(system: spins.SpinHamiltonianSystem, endianess: str = "little") -> qt.Q
...
+
+        """
+        number_spins: int
+        number_spins = spin_op.number_spins()
+        spin_operator = 0
+        coeff: complex
```
Comment (kbarkhqs) [inline struqture_qutip_interface/qutip_interface.py]: Can this type annotation be added to line 176?

File: tests/test_qutip_interface.py [hunk 5]
```diff
@@ -97,6 +98,22 @@ def test_qobj():
         sps.set(repr(pp), CalculatorComplex(i + 1))
     assert qi.qobj(sps) == qt.Qobj(exact, dims=[[2, 2], [2, 2]])
 
+def test_qobj2():
```
Comment (kbarkhqs) [inline tests/test_qutip_interface.py:101]: If you pass an empty SpinSystem, does the new code still work?

File: struqture_qutip_interface/qutip_interface.py (hunk 2 above)
Comment (dvy-gupta) [reply struqture_qutip_interface/qutip_interface.py]: Is the first more appropriate?

`spin_operator: qt.Obj = qt.Obj()`


`spin_operator = qt.Obj()`

File: tests/test_qutip_interface.py (hunk 5 above)
Comment (dvy-gupta) [reply tests/test_qutip_interface.py:101]: I added a test that creates an empty SpinSystem and compares it to an empty Qutip Object. Is that what you meant?

Comment (kbarkhqs) [reply tests/test_qutip_interface.py:101]: Yes that's perfect, thanks

File: struqture_qutip_interface/qutip_interface.py (hunk 2 above)
Comment (kbarkhqs) [reply struqture_qutip_interface/qutip_interface.py]: Yes that's perfect

**Review by kbarkhqs — APPROVED** [review]

**Pull request merged**